system-test/storage.ts

```diff
@@ -1024,9 +1027,22 @@ describe('storage', () => {
         await setUniformBucketLevelAccess(bucket, true);
         await setUniformBucketLevelAccess(bucket, false);
 
-        const [aclAfter] = await bucket.acl.default.get();
-        assert.deepStrictEqual(aclAfter, aclBefore);
-      });
+        // Setting uniform bucket level access is eventually consistent and may take up to a minute to be reflected
+        return new Promise(resolve => {
+          (async () => {
+            async function checkBucketAcl() {
+              try {
+                const [aclAfter] = await bucket.acl.default.get();
+                assert.deepStrictEqual(aclAfter, aclBefore);
+                resolve();
+              } catch {
+                setTimeout(checkBucketAcl, UNIFORM_ACCESS_WAIT_TIME);
+              }
+            }
+            await checkBucketAcl();
+          })();
+        });
```

Comment:
An alternate approach:
```suggestion
while (true) {
  try {
    const [aclAfter] = await bucket.acl.default.get();
     assert.deepStrictEqual(aclAfter, aclBefore);
     break;
  } catch {
     await new Promise(res => setTimeout(res, UNIFORM_ACCESS_WAIT_TIME));
  }
}
```